Applications/spire/source/toolbar/toolbar_menu.cpp

```diff
@@ -0,0 +1,160 @@
+#include "spire/toolbar/toolbar_menu.hpp"
+#include <QColor>
+#include "spire/spire/dimensions.hpp"
+#include "spire/ui/ui.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace spire;
+
+toolbar_menu::toolbar_menu(const QString& title, QWidget* parent)
+    : QPushButton(title, parent),
+      m_items(new QMenu(this)),
+      m_empty_style(true) {
+  setMenu(m_items);
+  connect(m_items, &QMenu::triggered, this, &toolbar_menu::on_triggered);
+  m_empty_item = new QWidgetAction(this);
+  m_empty_item->setText(tr("Empty"));
+  m_action_to_index[m_empty_item] = 0;
+  m_items->addAction(m_empty_item);
+    setStyleSheet(QString(R"(
+    QPushButton {
+      background-color: white;
+      border: 1px solid #C8C8C8;
+      font-family: Roboto;
+      font-size: %1px;
+      padding-left: %5px;
+      text-align: left;
+    }
+    QPushButton:focus, QPushButton:hover {
+      border: 1px solid #4B23A0;
+      outline: none;
+    }
+    QPushButton::menu-indicator {
+      image: url(":/icons/arrow-down.svg");
+      height: %3px;
+      padding-bottom: %6px;
+      padding-right: %2px;
+      width: %4px;
+    })")
+    .arg(scale_height(12)).arg(scale_width(8)).arg(scale_height(4))
+    .arg(scale_width(6)).arg(scale_width(8)).arg(scale_height(10)));
+
+  set_empty_menu_stylesheet();
+}
+
+void toolbar_menu::add(const QString& text) {
+  auto action = new QWidgetAction(this);
+  action->setText(text);
+  remove_empty_item();
+  if(m_empty_style) {
+    m_empty_style = false;
+    set_default_menu_stylesheet(scale_width(8));
+  }
+  m_items->addAction(action);
+  auto size = m_action_to_index.size();
```

Comment:
Don't need the size variable, just use the value directly;

  m_action_to_index[...] = m_action_to_index.size();